File: Tugas2/php/pc.php
<?php
include "disk.php";
include "proc.php";
include "ram.php";


class pc
{
	
    //Attribut
    private $procpc;
    private $diskpc;
    private $rampc;
	private $total = 0;
	
	
	//Construct
	public function __construct($procpc, $diskpc, $rampc){
		$this->procpc = $procpc;
		$this->diskpc = $diskpc;
		$this->rampc = $rampc;
        $this->total = $procpc->getpricep() + $diskpc->getpriced() + $rampc->getpricer();
	}
	
	//Set dan get
	public function setprocpc($procpc){
		$this->procpc = $procpc;
	}
	public function getprocpc(){
		return $this->procpc;
	}
	
	public function setdiskpc($diskpc){
		$this->diskpc = $diskpc;
	}
	public function getdiskpc(){
		return $this->diskpc;
	}
	
	public function setrampc($rampc){
		$this->rampc = $rampc;
	}
	public function getrampc(){
		return $this->rampc;
	}

    //Print Pc
    public function printpc(){
		echo "Procesor PC :". "<br/>";
        $this->procpc->printproc();
        echo "<br/>";
        echo "RAM PC :". "<br/>";
        $this->rampc->printram();
        echo "<br/>";
		echo "Disk PC :". "<br/>";
        $this->diskpc->printdisk();
        echo "<br/>";
        echo "Total Harga : ". $this->total;
	}
}
?>

File: Tugas1/pc.cpp
#include "proc.cpp"
#include "ram.cpp"
#include "disk.cpp"

using namespace std;

class pc{
	private:
		proc procpc;
        ram rampc;
        disk diskpc;
        int total;
	
	public:
		//Construct
		pc(){}
		
		pc(proc procpc, ram rampc, disk diskpc){
			this->procpc = procpc;
			this->rampc = rampc;
			this->diskpc = diskpc;
            this->total = procpc.getpricep() + rampc.getpricer() + diskpc.getpriced();
		}
		
		//Set dan Get
		void setprocpc(proc procpc)
        {
            this->procpc = procpc;
        }
		proc getprocpc(){
			return this->procpc;
		}
		
		void setrampc(ram rampc){
			this->rampc = rampc;
		}
		ram getrampc(){
			return this->rampc;
		}
		
		void setdiskpc(disk diskpc){
			this->diskpc = diskpc;
		}
		disk getdiskpc(){
			return this->diskpc;
		}

        int gettotal(){
            int x =  this->procpc.getpricep() + this->rampc.getpricer() + this->diskpc.getpriced();
            return x;
        }

        void output(){
            cout << "Procesor PC" << endl;
            this->procpc.printprosesor();
            cout << endl << "RAM PC" << endl;
            this->rampc.printram();
            cout << endl << "Disk PC" << endl;
            this->diskpc.printdisk();
            cout << endl << "Total Harga : " << this->gettotal() << endl;
        }
			
		~pc(){
		}
};